### ISSUE TITLE
After lock orientation view doesn't get rotated

### ISSUE THREAD

**Issue by vasyas** [expo/expo#4646]
## 🐛 Bug Report

```
  Expo CLI 2.18.4 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.7.0 - ~/.nvm/versions/node/v10.7.0/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.1.0 - ~/.nvm/versions/node/v10.7.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      @types/expo: ^32.0.13 => 32.0.13 
      @types/react: ^16.8.20 => 16.8.20 
      @types/react-native: ^0.57.63 => 0.57.63 
      @types/react-navigation: ^3.0.7 => 3.0.7 
      expo: ^33.0.0 => 33.0.5 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.0 => 3.11.0 
    npmGlobalPackages:
      expo-cli: 2.18.4
```

iOS 11, latest Expo client

When locking orientation to a different orientation than currently locked, screen doesn't get rotated.

For example, when locked to landscape orientation I do 
```
componentDidMount() {
  ScreenOrientation.lockAsync(OrientationLock.PORTRAIT);
}
``` 

My app is not get rotated. To see the app in the portrait mode, I need to rotate the phone to landscape and then back.

**Comment by cruzach** [comment]
Hi @vasyas,

I just created [this snack](https://snack.expo.io/@charliecruzan/screenorientationexample) and it seems to be working fine for me. Can you confirm that that doesn't work for you?

**Comment by Galzk** [comment]
@cruzach Hey, thanks for creating the snack, I can confirm that the issue is reproduced on my device, captured this video:
https://imgur.com/a/AcRMTNO

The orientation didn't change on its own, not even once (just to be clear)

I'm using a physical iPad Air 2, with software version 12.3.1
Updated to the latest Expo client available in the app store - 2.11.1

I can also confirm I can reproduce it on an iPad pro, latest Expo client as well

This is currently blocking us from releasing an update because our UI is dependant on that feature (long tables, on the same model iPad as mine, specifically used across our ORG to view data tailored for the screen size)

**Comment by cruzach** [comment]
Hey @Galzk , thanks for the video. I'm trying to reproduce this on my end, but keep seeing the correct behavior (even on iPad, same software).

Can you get this behavior in an Expo project on a simulator? Then maybe I can follow your steps to reproduce

**Comment by dylancompanjen** [comment]
Same here. Since the last update my screen is not rotated automatically anymore. I'm using lockAsync ALL on didMount and PORTRAIT on willUnmount. After my component unmounted I have to rotate the device by hand to get it back in PORTRAIT mode.

This used to work before updating to EXPO 33 and using the deprecated allowAsync.

**Comment by cruzach** [comment]
@dylancompanjen- could you provide a snack or repo where I can reproduce this? Like I said above, I'm getting the correct behavior in the snack I made. If you aren't, then let me know on what device, and with what source code

**Comment by dylancompanjen** [comment]
@cruzach all iOS devices. 
Our app uses PORTRAIT mode troughout all the app but lets you change the orientation when opening a modal that shows an image gallery.

On componentDidMount i'm using: ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.ALL);

on compontentWillUnmount I'm using: ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.PORTRAIT);

**Comment by cruzach** [comment]
Repro- https://snack.expo.io/@charliecruzan/screenorientationexample

When the app loads (should load into landscape), keep the device oriented in portrait. Click `lock portrait`, and nothing happens. Only after reorienting the device to landscape, and then back to portrait does the view lock into `portrait`.

**Comment by Galzk** [comment]
I've built an .ipa for testing our app with apple's testflight and I can confirm it's working as expected
We can narrow down the issue to the Expo Client, as working with an older version of our project on the newest Expo client version is broken as well, that wasn't the case before :)

**Comment by cruzach** [comment]
Hey @Galzk - i just tested this via TestFlight and saw the behavior there, too.

Can you make double check on your testflight app that you see it working? Is your code any different from the snack I linked?

**Comment by Galzk** [comment]
![carbon (3)](https://user-images.githubusercontent.com/34302982/61648750-bc064c00-acb8-11e9-9237-02a541eb28e1.png)

That's the whole code, seems pretty similar to the snack...
Few questions:

1. Did you check on a physical device?
2. Is any version different? iOS / expo?
3. Did you build the snack as it is?

I'm sure on our .ipa it's not happening, we've tested it for about half a day, and our clients would let us know 😅

**Comment by cruzach** [comment]
Hm not identical, you're simply unlocking rather than 
```
ScreenOrientation.lockAsync(
    ScreenOrientation.OrientationLock.PORTRAIT
);
```
Good to know!

**Comment by Galzk** [comment]
The thing is, the screen rotates to landscape and locks in this orientation...
If nothing happens in the demo, something else is broken IMO, let's try to understand what's up...
Would you like me to try on my physical device? If so, can you share with me the TestFlight?

**Comment by Hunsdale1** [comment]
I experienced the same issue @vasyas ... it isn't the most correct fix, I know, but I was able to get it to lock to portrait up ...

`componentWillUnmount = () => ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.PORTRAIT_UP)`

**Comment by Pitchlab** [comment]
It's not fixed in v36. At least, the behavior is different between iphone and ipad, and on ipad both portrait and landscape are not locked (ios 13.3 sim and device), but on iphone (sim) it works...

**Comment by cruzach** [comment]
@Pitchlab there's a separate issue for iPad screen orientation behavior in the Expo client here -> https://github.com/expo/expo/issues/5339

**Comment by yahyajamaldine** [comment]
The orientation didn't work for me too while trying to rotate the device, it has something to do with the configuration in the app.json file usually you finde something like this:
{
  "expo": {
    "name": "my-react",
    "slug": "my-react",
    "version": "1.0.0",
    "orientation": "portrait",
....
  }
}
-----------
what I did was removing the orientation property
--------------
{
  "expo": {
    "name": "my-react",
    "slug": "my-react",
 ....
  }
}

**Comment by lordprana** [comment]
I figured it out. In XCode, you need to find the Device Orientation under Deployment Info of the project.
<img width="883" alt="image" src="https://user-images.githubusercontent.com/5510659/153442630-5da028ff-da0d-4356-8a17-b3a4f10b159e.png">
